samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -166,9 +167,9 @@ def __enter__(self):
         :returns InvokeContext: Returns this object
         """
 
-        # Grab template from file and create a provider
-        self._template_dict = self._get_template_data(self._template_file)
-        self._function_provider = SamFunctionProvider(self._template_dict, self.parameter_overrides)
+        stacks = self._get_stacks()
+        self._template_dict = [stack for stack in stacks if stack.is_root_stack][0].template_dict
```

Comment:
Looks like we are using same function here, should we move finding root stack operation into a utility file? Which is `[stack for stack in stacks if stack.is_root_stack][0]`

Reply:
Done! thx